lib/ts/controllers/s-popover.ts

```diff
@@ -315,6 +324,177 @@ namespace Stacks {
+
+        if (controller) {
+            controller.hide();
+        } else if (isPopover) {
+            element.removeAttribute("data-s-popover-auto-show");
+            if (popover) {
+                popover.classList.remove("is-visible");
+            }
+        } else {
+            throw `element does not have data-controller="s-popover"`;
+        }
+    }
+
+    /**
+     * Options to use when attaching a popover via `Stacks.attachPopover`.
+     * @see Stacks.attachPopover
+     */
+     export interface PopoverOptions {
+        /**
+         * When true, the `click->s-popover#toggle` action will be attached to the controller element or reference element.
+         */
+       toggleOnClick?: boolean;
+        /**
+         * When set, `data-s-popover-placement` will be set to this value on the controller element.
+         */
+        placement?: string;
+
+        /**
+         * When true, the popover will appear immediately when the controller connects.
+         */
+        autoShow?: boolean;
+    }
+
+    /**
+     * Attaches a popover to an element and performs additional configuration.
+     * @param element the element that will receive the `data-controller="s-popover"` attribute.
+     * @param popover an element with the `.s-popover` class or HTML string containing a single element with the `.s-popover` class.
+     *                If the popover does not have a parent element, it will be inserted as a immediately after the reference element.
+     * @param options an optional collection of options to use when configuring the popover.
+     */
+     export function attachPopover(element: Element, popover: Element | string, options?: PopoverOptions)
+     {
+        const { referenceElement, popover: existingPopover } = getPopover(element);
+
+        if (existingPopover) {
+            throw `element already has popover with id="${existingPopover.id}"`
+        }
+
+        if (!referenceElement) {
+            throw `element has invalid data-s-popover-reference-selector attribute`
+        }
+
+        if (typeof popover === 'string') {
+            const elements = document.createRange().createContextualFragment(popover).children;
+            if (elements.length !== 1) {
+                throw "popover should contain a single element";
+            }
+            popover = elements[0];
+        }
+
+        const existingId = referenceElement.getAttribute("aria-controls");
+        var popoverId = popover.id;
+
+        if (!popover.classList.contains('s-popover')) {
+            throw `popover should have the "s-popover" class but had class="${popover.className}"`;
+        }
+
+        if (existingId && existingId !== popoverId) {
+            throw `element has aria-controls="${existingId}" but popover has id="${popoverId}"`;
+        }
+
+        if (!popoverId) {
+            popoverId = "--stacks-s-popover-" + Math.random().toString(36).substring(2, 10);
```

Comment:
We have a similar function in `TooltipController.generateId()`.  Maybe we could add a base method on StacksController:

```
export function generateId() {
    return `--stacks-${this.identifier}-{Math.random().toString(36).substring(2, 10)}`;
}
```

Reply:
I'm punting on this one for now. I'd have to make the method static (since it is accessed in a helper here), in which case it won't have access to `this.identifier` and I'd have to hardcode the controller name somewhere for both types